transcom/mymove
# B 17950 convert to customer expense

## [Agility ticket](B-[17950](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A869423))

## Summary

Adds a "Convert to Customer's Expense" checkbox when a TOO chooses to deny a SIT extension request on an origin 1st day SIT item in the "Move Task Order" tab of a move.

When the box is checked and the form is submitted, there will be a flag in the database, both in `sit_extensions` and `mto_service_items` to denote that this item has been converted and the service member is responsible for payment. Future PR's will add UI functionality that will use these flags to alert the user of the change.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Login as TOO
2. Access/create a move with an HHG shipment and a 1st day origin SIT service item.
3. Get the shipment ID (not move ID) by either using devtools or login as Prime, select the move, and copy the shipment ID from the service items list.
4. Using Postman, hit the Prime API endpoint at mto-shipments/{mtoShipmentID}/sit-extensions and POST with a request body like:
`{
  "requestReason": "SERIOUS_ILLNESS_MEMBER",
  "contractorRemarks": "REMARK",
  "requestedDays": 30
}`
5. Now you should see "additional days requested" as a TOO on the move task order screen. Click "Review Request"
6. You can choose to accept or reject, if reject is selected, Convert to Customers Expense should show as a checkbox, and a remark should be required before Submit can be clicked.
7. After submit, since no UI is implemented, you'll have to check the database with a tool like DBeaver to check both the sit_extensions table and mto_service_items table. You should see the last column `members_expense` marked as TRUE for the shipment item/extension request.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
